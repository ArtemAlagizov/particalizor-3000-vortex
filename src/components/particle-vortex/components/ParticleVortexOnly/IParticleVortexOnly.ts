export interface IParticleVortexOnly {
  particleNumberValue: number;
  particleTraceWidthValue: number;
  particleLifeTimeValue: number;
  backgroundColor: string;
  vortexNumberValue: number;
}
