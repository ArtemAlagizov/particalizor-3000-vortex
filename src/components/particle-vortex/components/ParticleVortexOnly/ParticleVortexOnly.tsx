import React from "react";
import { IParticleVortexOnly } from "./IParticleVortexOnly";
import { ParticleVortex } from "particalizor-3000";
import ContainerDimensions from "react-container-dimensions";

const ParticleVortexOnly: React.FC<IParticleVortexOnly> = ({
  particleNumberValue,
  particleTraceWidthValue,
  particleLifeTimeValue,
  backgroundColor,
  vortexNumberValue
}) => {
  return (
    <ContainerDimensions>
      {({ height, width }) => (
        <ParticleVortex
          imageWidth={width}
          imageHeight={height}
          vortexNumber={vortexNumberValue}
          particleTraceWidth={particleTraceWidthValue}
          particleNumber={particleNumberValue}
          particleLifeTime={particleLifeTimeValue}
          backgroundColor={backgroundColor}
        />
      )}
    </ContainerDimensions>
  );
};

export default React.memo(ParticleVortexOnly);
