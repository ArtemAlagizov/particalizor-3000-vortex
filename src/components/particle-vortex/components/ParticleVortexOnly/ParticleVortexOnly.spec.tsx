import React from "react";
import { shallow } from "enzyme";
import ParticleVortexOnly from "./ParticleVortexOnly";

describe("Given a ParticleVortexOnly", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <ParticleVortexOnly
          particleTraceWidthValue={1}
          particleNumberValue={1}
          particleLifeTimeValue={1}
          backgroundColor={"red"}
          vortexNumberValue={3}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
