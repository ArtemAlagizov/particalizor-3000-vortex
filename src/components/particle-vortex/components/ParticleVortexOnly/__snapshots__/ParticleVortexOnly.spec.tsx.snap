// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ParticleVortexOnly when rendering should match snapshot 1`] = `
<ContainerDimensions>
  <Component />
</ContainerDimensions>
`;
